source/framework/core/src/TRestHits.cxx

```diff
@@ -546,28 +546,165 @@ Double_t TRestHits::GetSigmaY() {
 
     return sigmaY = TMath::Sqrt(sigmaY2);
 }
-
-Double_t TRestHits::GetGaussSigmaX(Int_t readoutChannels, Int_t startChannel, Int_t endChannel, Double_t pitch) {
+/**
+Double_t TRestHits::GetGaussSigmaX(Int_t nBins = 100) {
 	TH1::AddDirectory(kFALSE);
     Double_t gausSigmaX = 0;
 
-	TH1D* hX = new TH1D("hitsGaussHistoX", "hitsGaussHistoX", readoutChannels, startChannel, endChannel);
+     //double min = *min_element(fX.begin(), fX.end());
+     //double max = *max_element(fX.begin(), fX.end());
+
 
+   /*
+	TH1D* hX = new TH1D("hitsGaussHistoX", "hitsGaussHistoX", readoutChannels, startChannel, endChannel);
+	//hX->Sumw2();
     for (int n = 0; n < GetNumberOfHits(); n++) hX->Fill(fX[n], fEnergy[n]);
+	*//**
+	for (int n = 0; n < GetNumberOfHits(); n++) {
+		hX->Fill(fX[n], fEnergy[n]);
+		cout << "fX[n] is " << fX[n] << endl;
+		cout << "energy is: " << fEnergy[n] << endl;
+	}
+	**/
+	//WriteHitToTextFile("/home/cristina/hits");
+   	/*TCanvas *c = new TCanvas("c","X position fit",200,10,500,500);
+	hX->Draw();
 
     TF1* fit = new TF1("", "gaus", hX->GetMaximumBin() / (1/pitch) - 32, hX->GetMaximumBin() / (1/pitch) - 26);
 
-    hX->Fit(fit, "QNRL");  // Q = quiet, no info in screen; N = no plot; R = fit in the function range; L = log likelihood fit
+    hX->Fit(fit, "QRL");  // Q = quiet, no info in screen; N = no plot; R = fit in the function range; L = log likelihood fit (default is chi-square) ; W = Set all weights to 1 for non empty bins; ignore error bars; WW = Set all weights to 1 including empty bins; ignore error bars
+	
+	//cout << "Drawing fit" << endl;	
+	//fit->Draw();
+
+	c->Update();
 
     gausSigmaX = fit->GetParameter(2);
 
-    delete fit;
-	delete hX;
+    //delete fit;
+	//delete hX;
 
 	TH1::AddDirectory(kTRUE);
+    
     return gausSigmaX;
-}
-
+     */
+    //return max;
+//}
+/**
+void TRestHits::WriteHitToTextFile(TString filename) {
```

Comment:
This function is commented here, but it is defined at the header. That will cause problems at some point. But I dont see why it is connected. It might be handy as a way to export and debug hits information.

Just the name, could be better in plural `WriteHitsToTextFile`, and making it a bit more generic, by writting X,Y,Z. However, I would add a conditional so that if the hit type is XZ, write a NaN at the text file.